src/stream/src/executor/source/legacy_fs_source_executor.rs

```diff
@@ -271,10 +275,7 @@ impl<S: StateStore> LegacyFsSourceExecutor<S> {
             core.split_state_store.set_all_complete(completed).await?
         }
         // commit anyway, even if no message saved
-        core.split_state_store
-            .state_table
-            .commit_assert_no_update_vnode_bitmap(epoch)
-            .await?;
+        core.split_state_store.commit(epoch).await?;
```

Comment:
So we are committing the offset after meta has collected current epoch's barrier?

Reply:
Here is just a refactor, to change the method name we call and no original logic is changed here. 

The correct order should be: `receive barrier` -> `commit state store` -> `yield barrier` -> `meta collect barrier`. The current code is following this correct order.